File: src/youtube_automation/utils/config/localizations.py
"""ローカライゼーション設定の責務別 dataclass."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass(frozen=True)
class Localizations:
    """`localizations.json`（`config/` 直下、`config/channel/` の外）の内容.

    - `exists=True` 時: `data` には JSON 全量、`supported_languages` / `default_language` も埋める
    - `exists=False` 時: `data={}`、`supported_languages=[youtube.api.language]`、`default_language=""`
    """

    data: dict = field(default_factory=dict)
    exists: bool = False
    supported_languages: list[str] = field(default_factory=list)
    default_language: str = ""

File: packages/core/src/config/localizations.ts
// ローカライゼーション設定（`config/localizations.json`、`config/channel/` の外の別ファイル）。
//
// - exists=true: `data` に JSON 全量（raw passthrough）、`supportedLanguages` /
//   `defaultLanguage` も埋める
// - exists=false: `data={}`、`supportedLanguages=[youtube.api.language]`、`defaultLanguage=""`

import { z } from "zod";

/** localizations.json の内容を parse して計算済みフィールドへ transform する。 */
export const Localizations = z
  .looseObject({
    default_language: z.string().default(""),
    supported_languages: z.array(z.string()).default([]),
  })
  .transform((o) => ({
    data: o as Record<string, unknown>,
    defaultLanguage: o.default_language,
    exists: true,
    supportedLanguages: o.supported_languages,
  }));

export type Localizations = z.infer<typeof Localizations>;

/** localizations.json が存在しないチャンネルのフォールバック値。 */
export const localizationsAbsent = (
  fallbackLanguage: string
): Localizations => ({
  data: {},
  defaultLanguage: "",
  exists: false,
  supportedLanguages: [fallbackLanguage],
});
